### PR TITLE
CB2-7557 Fix for correct timestamp handling

### PR DESCRIPTION
**Pull request by ChrisPeska-BJSS**
## Description

Timestamp related attributes in Technical Records and Test Results are predominantly of ISO 8601 format which is 24 or 27 characters long (YYYY-MM-DDTHH:mm:ss.sssZ or ±YYYYYY-MM-DDTHH:mm:ss.sssZ). The timezone is always zero UTC offset, as denoted by the suffix Z. Timestamp values are parsed to `yyyy-mm-dd hh24:mi:ss.fff` string format and upserted into database.

## Issue

A helper function `getDate()` was converting `.fff` fractions of second incorrectly, without left-side padding. Values in the database are padded to their maximum precision, e.g.

input value : `"2021-05-18T09:14:41.009Z"` -> was parsed to `2021-05-18 09:14:41.9`
upseted value : `"2021-05-18T09:14:41.900Z"` (once inserted into the database)

tldr; 
Fix for handling string DateTime/Timestamps correctly.

Related issue: [CB2-7563](https://dvsa.atlassian.net/browse/CB2-7563)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

**Review by Ian-Hodges — DISMISSED** [review]

File: src/utils/padwithzeros.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function padWithZeros(num: number, maxSize: number): string {
```
Comment (mcadenas-bjss) [inline src/utils/padwithzeros.ts:1]: This looks to me like it could be done with `String.prototype.padStart()`?
e.g. `return num.toString().padStart(maxSize, 0);`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

**Review by mcadenas-bjss — APPROVED** [review]

File: src/utils/padwithzeros.ts (hunk 1 above)
Comment (ChrisPeska-BJSS) [reply src/utils/padwithzeros.ts:1]: Well spotted, thank you @mcadenas-bjss 👍

**Review by Ian-Hodges — APPROVED** [review]

**Review by mcadenas-bjss — APPROVED** [review]
